src/vm/dllimport.cpp

```diff
@@ -6132,22 +6136,107 @@ bool         NDirect::s_fSecureLoadLibrarySupported = false;
 #endif // !FEATURE_PAL
 
 // static
-NATIVE_LIBRARY_HANDLE NDirect::LoadLibraryFromPath(LPCWSTR libraryPath)
+NATIVE_LIBRARY_HANDLE NDirect::LoadLibraryFromPath(LPCWSTR libraryPath, BOOL throwOnError)
 {
     STANDARD_VM_CONTRACT;
 
     LoadLibErrorTracker errorTracker;
     const NATIVE_LIBRARY_HANDLE hmod =
         LocalLoadLibraryHelper(libraryPath, GetLoadWithAlteredSearchPathFlag(), &errorTracker);
```

Comment:
I dont' think this matches the semantic of the callers. Altered search path looks in a bunch of different places and isn't secure.

Reply:
The [only difference](https://docs.microsoft.com/en-us/windows/desktop/Dlls/dynamic-link-library-search-order) caused by the LOAD_WITH_ALTERED_SEARCH_PATH is: "The standard search begins in the calling application's directory, and the alternate search begins in the directory of the executable module that LoadLibraryEx is loading."

I kept the LOAD_WITH_ALTERED_SEARCH_PATH in this case because it was already used by the AssemblyLoadContext.LoadUnmanagedDll() implementation. I think that these two APIs should at least agree on what the default flags are (hence they use the same method for loading).

We can pick the default without LOAD_WITH_ALTERED_SEARCH_PATH for both Marshall.LoadLibrary() and AssemblyLoadContext.LoadUnmanagedDll() if we think that is better. I'm not really sure if one is clearly better. But I think we should make this change separately since it modifies existing behavior.

Note that when we try to load from NATIVE_DLL_SEARCH_DIRECTORIES (given  by the host), we also LOAD_WITH_ALTERED_SEARCH_PATH https://github.com/dotnet/coreclr/blob/master/src/vm/dllimport.cpp#L6416

Reply:
When DllImportSearchPathFlags are absent (as in this case), the documented behavior is to use LOAD_WITH_ALTERED_SEARCH_PATH.

"The common language runtime handles the call to the LoadLibraryEx function according to the following algorithm:

If the attribute is applied to an individual platform invoke, use the values specified by that instance of the attribute.

Otherwise, if the attribute is applied to the assembly that contains the platform invoke, use the values specified by that instance of the attribute.

Otherwise, search the assembly directory and then call the LoadLibraryEx function with the LOAD_WITH_ALTERED_SEARCH_PATH flag.
"